sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/samples/TimeSeriesInsightsClientSample/HierarchiesSamples.cs

```diff
@@ -0,0 +1,158 @@
+using Azure.IoT.TimeSeriesInsights.Models;
+using static Azure.IoT.TimeSeriesInsights.Samples.SampleLogger;
+
+namespace Azure.IoT.TimeSeriesInsights.Samples
+{
+    internal class HierarchiesSamples
+    {
+        /// <summary>
+        /// This sample demonstrates usage of Time Series Insights hierarchy APIs.
+        /// </summary>
+        public async Task RunSamplesAsync(TimeSeriesInsightsClient client)
+        {
+            // For the purpose of keeping code snippets readable to the user, hardcoded string literals are used in place of assigned variables, eg Ids.
+            // Despite not being a good code practice, this prevents code snippets from being out of context for the user when making API calls that accept Ids as parameters.
+
+            PrintHeader("TIME SERIES INSIGHTS HIERARCHIES SAMPLE");
+
+            #region Snippet:TimeSeriesInsightsSampleCreateHierarchies
+            var tsiHierarchyName = "sampleHierarchy";
+            var tsiInstanceField1 = "hiearchyLevel1";
+            var hierarchySource = new TimeSeriesHierarchySource();
+            hierarchySource.InstanceFieldNames.Add(tsiInstanceField1);
+
+            var tsiHierarchy = new TimeSeriesHierarchy(tsiHierarchyName, hierarchySource);
+            tsiHierarchy.Id = "sampleHiearchyId";
+
+            var timeSeriesHierarchies = new List<TimeSeriesHierarchy>
+            {
+                tsiHierarchy
+            };
+
+            // Create Time Series hierarchies
+            Response<TimeSeriesHierarchyOperationResult[]> createHierarchiesResult = await client
+                .Hierarchies
+                .CreateOrReplaceAsync(timeSeriesHierarchies)
+                .ConfigureAwait(false);
+
+            // The response of calling the API contains a list of error objects corresponding by position to the input parameter array in the request.
+            // If the error object is set to null, this means the operation was a success.
+            for (int i = 0; i < createHierarchiesResult.Value.Length; i++)
+            {
+                if (createHierarchiesResult.Value[i].Error == null)
+                {
+                    Console.WriteLine($"Created Time Series hierarchy successfully.");
+                }
+                else
+                {
+                    Console.WriteLine($"Failed to create a Time Series hieararchy: {createHierarchiesResult.Value[i].Error.Message}.");
+                }
+            }
+            #endregion Snippet:TimeSeriesInsightsSampleCreateHierarchies
+
+            #region Snippet:TimeSeriesInsightsSampleGetAllHiearchies
+            // Get all Time Series hierarchies in the environment
+            AsyncPageable<TimeSeriesHierarchy> getAllHierarchies = client.Hierarchies.GetAsync();
+            await foreach (TimeSeriesHierarchy hierarchy in getAllHierarchies)
+            {
+                Console.WriteLine($"Retrieved Time Series Insights hierarchy with Id: '{hierarchy?.Id}' and Name: '{hierarchy?.Name}'");
```

Comment:
why would hierarchy be null? Is that a possibility?

Reply:
Good observation. Couldn't recall a reason why I added this check, will fix.